webapp/channels/src/components/new_search/search_hint.tsx

```diff
@@ -0,0 +1,101 @@
+
+import React from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import styled from 'styled-components';
+
+import {searchHintOptions, searchFilesHintOptions} from 'utils/constants';
+
+type Props = {
+    onSelectFilter: (filter: string) => void;
+    searchType: string;
+    searchTerms: string;
+    hasSelectedOption: boolean;
+    isDate: boolean;
+}
+
+const SearchHintsContainer = styled.div`
+    display: flex;
+    padding: 20px 24px;
+    color: rgba(var(--center-channel-color-rgb), 0.75);
+    i {
+        margin-right: 8px;
+        color: var(--center-channel-color-56);
+    }
+`;
+
+const SearchFilter = styled.button`
+    display: flex;
+    padding: 4px 10px;
+    color: var(--center-channel-color);
+    background: rgba(var(--center-channel-color-rgb), 0.08);
+    border-radius: var(--radius-l);
+    border: none;
+    font-size: 10px;
+    font-weight: 600;
+    line-height: 12px;
+    margin-left: 8px;
+    &:hover {
+        background: rgba(var(--center-channel-color-rgb), 0.16);
+    }
+`;
+
+const SearchHints = ({onSelectFilter, searchType, searchTerms, hasSelectedOption, isDate}: Props): JSX.Element => {
+    const intl = useIntl();
+    let filters = searchHintOptions.filter((filter) => filter.searchTerm !== '-' && filter.searchTerm !== '""');
+    if (searchType === 'files') {
+        filters = searchFilesHintOptions.filter((filter) => filter.searchTerm !== '-' && filter.searchTerm !== '""');
+    }
+
+    if (isDate) {
+        return <></>;
+    }
+
+    if (hasSelectedOption) {
+        return (
+            <SearchHintsContainer id='searchHints'>
+                <i className='icon icon-keyboard-return'/>
+                <FormattedMessage
+                    id='search_hint.enter_to_select'
+                    defaultMessage='Press Enter to select'
+                />
+            </SearchHintsContainer>
+        );
+    }
+
+    if (searchTerms.length > 0 && searchTerms[searchTerms.length - 1] !== ' ') {
+        return (
+            <SearchHintsContainer id='searchHints'>
+                <i className='icon icon-keyboard-return'/>
+                <FormattedMessage
+                    id='search_hint.enter_to_search'
+                    defaultMessage='Press Enter to search'
+                />
+            </SearchHintsContainer>
+        );
+    }
+
+    return (
+        <SearchHintsContainer id='searchHints'>
+            <i className='icon icon-lightbulb-outline'/>
+            <FormattedMessage
+                id='search_hint.filter'
+                defaultMessage='Filter your search with:'
+            />
+            {filters.map((filter) => (
+                <SearchFilter
+                    key={filter.searchTerm}
+                    onClick={() => onSelectFilter(filter.searchTerm)}
+                >
+                    <span title={intl.formatMessage(filter.message)}>
+                        {filter.searchTerm}
+                    </span>
```

Comment:
I don't think we need this inner span because `title` isn't a valid attribute for a `span`

Reply:
title, according to Mozilla page is a global attribute, so also applies to span (https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes), And this is helping me to show the description of the different filters.

Reply:
Oh right, I confused that with the `alt` attribute since I've seen some other places where people tried to add that to a `span`